backend/src/main/java/com/woowacourse/momo/auth/dto/TokenInfo.java

```diff
@@ -0,0 +1,4 @@
+package com.woowacourse.momo.auth.dto;
+
+public record TokenInfo(long id, String nickname) {
```

Comment:
이 부분은 조금 생각이 다른데요, `TokenInfo`가 생성된 이상 `id`필드는 `nullable` 하지 않음을 사용자에게 확실히 알려줄 수 있어야 한다고 생각합니다.

오히려 `TokenInfo`를 인스턴스화 하는 문맥에서 혹시 `id`로 전달하는 값이 `null`이지는 않은지 검증하고, 생성된 이후에는 항상 유효한 `id` 값이 들어있음을 보장받을 수 있어야 해요.

그렇지 않다면 토큰 인스턴스를 사용할 때 마다 `null` 체크를 해야 하는 불편함이 생기지 않을까요?
영속화되지 않은 엔티티와 같이 `id` 필드가 비어있어야 하는 특별한 문맥이 존재하는 게 아니라면 객체의 식별자로 사용될 필드는 항상 원시타입으로 가져가는 게 좋은 것 같습니다!

---
현재 구현에서 `TokenInfo` 생성 시 `id`에 `null`이 전달되면 `0`으로 초기화되는 맥락을 공유해 주실 수 있으실까요?

아래 테스트에서는 테스트 시점(45행)이 아닌 생성 시점(44행)에서 NPE가 발생합니다!
<img width="525" alt="image" src="https://github.com/user-attachments/assets/7b7d6888-5f6e-41d4-bbed-760b60e6cf0f">
```java
Cannot invoke "java.lang.Long.longValue()" because "id" is null
java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because "id" is null
	at com.woowacourse.momo.domain.attendee.AttendeeTest.nullUnBoxingTest(AttendeeTest.java:44)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

```

Reply:
저도 페드로의 의견과 같고, 토큰을 검증하는 과정에 `NPE`를 고려할 것이라면 `verifyToken()` 메서드에서 `NPE`를 catch 하는게 맞다는 생각입니다.

그런데 제 추측이지만 만약 토큰이 훼손되었다면 `TokenInfo`를 생성하기 이전에 토큰을 검증하는 과정에서 이미 catch한 에러가 발생할 것 같습니다.

Reply:
> 현재 구현에서 TokenInfo 생성 시 id에 null이 전달되면 0으로 초기화되는 맥락을 공유해 주실 수 있으실까요?

TokenInfo가 record라서 token필드에 final을 명시하지 않아도 기본으로 final이 된다는 점을 제가 놓쳤네요🥲
만약 final이 아니라면, `new Tokeninfo(String nickname)`라는 생성자를 사용할 수도 있고, 그렇게 되면 0으로 초기화 될 수도 있겠다고 생각했어요..! record에 익숙하지 못해서 그렇게 생각했던 것 같습니다..ㅎ

Reply:
>이 부분은 조금 생각이 다른데요, TokenInfo가 생성된 이상 id필드는 nullable 하지 않음을 사용자에게 확실히 알려줄 수 있어야 한다고 생각합니다.

오히려 TokenInfo를 인스턴스화 하는 문맥에서 혹시 id로 전달하는 값이 null이지는 않은지 검증하고, 생성된 이후에는 항상 유효한 id 값이 들어있음을 보장받을 수 있어야 해요.

그렇지 않다면 토큰 인스턴스를 사용할 때 마다 null 체크를 해야 하는 불편함이 생기지 않을까요?
영속화되지 않은 엔티티와 같이 id 필드가 비어있어야 하는 특별한 문맥이 존재하는 게 아니라면 객체의 식별자로 사용될 필드는 항상 원시타입으로 가져가는 게 좋은 것 같습니다!

동의합니다~!! 좋아요👍